packages/web/src/components/import-export-srcbook-modal.tsx

```diff
@@ -86,33 +284,149 @@ export function ExportSrcbookModal({
+          break;
+      }
+    }, 0);
+  }, [activeTab]);
+
+  // When hiding the modal, reset all controls
+  useEffect(() => {
+    if (!open) {
+      return;
+    }
+
+    setActiveTab('file');
+  }, [open]);
+
+  const downloadFileName = useMemo(() => {
+    return `${getTitleForSession(session)}.src.md`;
+  }, [session]);
+
+  function onDownloadSrcbook() {
+    if (srcbookText.status !== 'complete') {
+      return;
+    }
+
+    const blob = new Blob([srcbookText.text]);
+
+    const url = URL.createObjectURL(blob);
+    const link = document.createElement('a');
+    link.setAttribute('href', url);
+    link.setAttribute('download', downloadFileName);
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+    URL.revokeObjectURL(url);
+  }
+
+  function onCopySrcbookToClipboard() {
+    if (srcbookText.status !== 'complete') {
+      return;
     }
+
+    onOpenChange(false);
+    navigator.clipboard.writeText(srcbookText.text);
+    toast.success('Copied to clipboard.');
   }
 
   return (
     <Dialog open={open} onOpenChange={onOpenChange}>
       <DialogContent className="max-w-3xl">
         <DialogHeader>
-          <DialogTitle>Save to file</DialogTitle>
+          <DialogTitle>Export Srcbook</DialogTitle>
           <DialogDescription asChild>
             <p>
               Export this Srcbook to a <code className="code">.src.md</code> file which is shareable
               and can be imported into any Srcbook application.
             </p>
           </DialogDescription>
         </DialogHeader>
-        <ExportLocationPicker onSave={onSave} title={getTitleForSession(session)} />
-        {error && <p className="text-destructive-foreground">{error}</p>}
+
+        <Tabs value={activeTab} onValueChange={(tab) => setActiveTab(tab as 'file' | 'text')}>
+          <div className="border-b mb-4">
+            <TabsList>
+              <TabsTrigger value="file">
+                <div className="flex items-center gap-2">
+                  <FilesIcon size={18} />
+                  To Filesystem
+                </div>
+              </TabsTrigger>
+              <TabsTrigger value="text">
+                <div className="flex items-center gap-2">
+                  <NotebookIcon size={18} />
+                  To Text
+                </div>
+              </TabsTrigger>
+            </TabsList>
+          </div>
+
+          <TabsContent className="mt-0" value="file">
+            <button
+              className="flex flex-col gap-4 w-full items-center justify-center h-[160px] border border-dashed rounded-md cursor-default"
+              onClick={onDownloadSrcbook}
+            >
+              <div className="flex flex-col items-center gap-2 text-tertiary-foreground">
+                <NotebookIcon size={24} />
+                <code className="text-xs">{downloadFileName}</code>
+              </div>
+              <Button>Download File</Button>
+            </button>
```

Comment:
Note that there is a slight bit of functionality being lost here - previously, users could save a `*.src.md` file anywhere on their filesystem. Now, when it is downloaded, it only ends up in their downloads directory. Unless I'm missing something (and without reaching for [chrome specific apis](https://developer.chrome.com/docs/capabilities/web-apis/file-system-access)) I'm not sure there's a way around this.

Reply:
I'm not an expert with these newer APIs, but the [showSaveFilePicker](https://developer.mozilla.org/en-US/docs/Web/API/Window/showSaveFilePicker) function seems to be what we want. It may not work in all browsers, but if it works in chrome and edge that should solve for at least 80%, if not 95%+.

Perhaps we can check for its existence and use it, otherwise fallback to what we have in this PR?

Reply:
@versecafe is there a reason we can't we use it for browsers that support it and then fallback to what is here for browsers that don't?

The vast majority of people use chrome, so as the comment above states I would expect this to cover almost everyone. I'm not sure what ladybird is but a google search says "It is planning to have an alpha release in 2026" so I'm not too concerned about this affecting anyone

Reply:
To close the loop on this one, here's what I did to use the web filesystem api while falling back to the old download behavior if this doesn't exist: https://github.com/srcbookdev/srcbook/pull/307/commits/ec54805f0c7f771c8990d39695b08b0ecaa37dd0